include/Smarty/plugins/function.sugar_include.php

```diff
@@ -78,14 +79,22 @@ function smarty_function_sugar_include($params, &$smarty)
 	   	  foreach($params['include'] as $include) {
 	   	  	      if(isset($include['file'])) {
 	   	  	         $file = $include['file'];
-	   	  	         if(preg_match('/[\.]js$/si',$file)) {
-	   	  	            $code .= "<script src=\"". getJSPath($include['file']) ."\"></script>";
-	   	  	         } else if(preg_match('/[\.]php$/si', $file)) {
+                     if (preg_match('/tiny_mce.*[\.]js$/si', $file)) {
```

Comment:
Why tiny_mce files are exceptionals?

Reply:
We have tiny_mce files in two editviewdefs.php:

```
modules/AOK_KnowledgeBase/metadata/editviewdefs.php:          'file' => 'include/javascript/tiny_mce/tiny_mce.js',
modules/AOS_PDF_Templates/metadata/editviewdefs.php:          'file' => 'include/javascript/tiny_mce/tiny_mce.js',
```

And Tiny MCE stops working when you concatenate it to another minyfied JS file.... I spent many hours debugging it and I really cannot find the reason of this behavior ... I find this in my first commit because a selenium test fails.

For this I decided to treat this files in an exceptional way... besides tiny_mce files are already minyfied so it is not so bad

Reply:
We are using it even before initial commit... That comment is there since 06/06/2012, commit c91080dc82 in  SugarCRM CE 6.5.0 code! :-0


```bash
/m/d/t/gca git:release/19 ❯❯❯ git blame jssource/jsmin.php | head                                                                                                                                                                                            
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  1) <?php
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  2)
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  3) class SugarMin {
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  4)
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  5)     /**
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  6)      * jsParser will take javascript source code and minify it.
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  7)      *
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  8)      * Note: There is a lot of redundant code since both passes
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400  9)      * operate similarly but with slight differences. It will probably
c91080dc82 (John Mertic 2012-06-06 20:57:07 -0400 10)      * be a good idea to refactor the code at a later point when it is stable.
```

I get your point about the fact that that code is not stable...but we are using that code every day when we execute "Repair JS Files" from the Administration module. 

Anyway I agree with you, we should refactor that code. 
What about using that code for this PR, and refactor that code in another PR? After all we are living with that code since always, and I think there were no big issues with it.

Reply:
Hi @gymad 

> The minified version of js files should doing exactly same as the original, even if the Minifier class works as expected

About TinyMCE and why I treated its files in an exceptional way: I was debugging and find that the problem is that TinyMCE load plugins (.js files) on the fly.

The following screenshot is when the code of this PR do not minify and concatenate tiny_mce.js file:

![tiny_mce_ok](https://user-images.githubusercontent.com/939888/50709402-76e53a00-1046-11e9-9f32-8addbabddca2.png)

As you can see, the plugins are loaded on the fly ok.

But If we minify and concatenate tiny_mce.js file (and thus change the name and location of the resultant js file), the following happens:

![tiny_mce_nook](https://user-images.githubusercontent.com/939888/50709514-d6434a00-1046-11e9-9977-ea28bdccf984.png)

As you can see in the last screenshot, TinyMCE fails to load the plugins on the fly, because fails in discovering the right path to the plugins files.

I think this is a TinyMCE issue not something wrong in joinAndMinifyJSFiles function.